src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figuresArray = new Figure[6];
+        FigureSupplier figureSupplier = new FigureSupplier();
+        for (int i = 0; i < figuresArray.length; i++) {
+            if (i > Math.floor(figuresArray.length / 2)) {
```

Comment:
To enhance readability and reduce the risk of errors, consider using a constant for the threshold value instead of calculating it inline. Additionally, using Math.floor is unnecessary when dividing integers, as the result is already an integer division.